pinot-core/src/main/java/org/apache/pinot/core/query/optimizer/statement/JsonStatementOptimizer.java

```diff
@@ -0,0 +1,561 @@
+            String[] parts = getIdentifierParts(left.getIdentifier());
+            if (parts.length > 1 && isValidJSONColumn(parts[0], schema)) {
+              if (isIndexedJSONColumn(parts[0], config)) {
+                Function jsonMatchFunction = new Function("JSON_MATCH");
+
+                List<Expression> jsonMatchFunctionOperands = new ArrayList<>();
+                jsonMatchFunctionOperands.add(RequestUtils.createIdentifierExpression(parts[0]));
+                jsonMatchFunctionOperands.add(RequestUtils.createLiteralExpression(new StringLiteralAstNode(
+                    getJsonPath(parts, true) + getOperatorSQL(kind) + getLiteralSQL(right.getLiteral(), false))));
+                jsonMatchFunction.setOperands(jsonMatchFunctionOperands);
+
+                expression.setFunctionCall(jsonMatchFunction);
+              } else {
+                left.clear();
+                left.setType(ExpressionType.FUNCTION);
+                left.setFunctionCall(getJsonExtractFunction(parts, getColumnTypeForLiteral(right.getLiteral())));
+              }
+            }
+          }
+          break;
+        }
+        case IS_NULL:
+        case IS_NOT_NULL: {
+          Expression operand = operands.get(0);
+          if (operand.getType() == ExpressionType.IDENTIFIER) {
+            String[] parts = getIdentifierParts(operand.getIdentifier());
+            if (parts.length > 1 && isValidJSONColumn(parts[0], schema)) {
+              if (isIndexedJSONColumn(parts[0], config)) {
+                Function jsonMatchFunction = new Function("JSON_MATCH");
+
+                List<Expression> jsonMatchFunctionOperands = new ArrayList<>();
+                jsonMatchFunctionOperands.add(RequestUtils.createIdentifierExpression(parts[0]));
+                jsonMatchFunctionOperands.add(RequestUtils.createLiteralExpression(
+                    new StringLiteralAstNode(getJsonPath(parts, true) + getOperatorSQL(kind))));
+                jsonMatchFunction.setOperands(jsonMatchFunctionOperands);
+
+                expression.setFunctionCall(jsonMatchFunction);
+              } else {
+                operand.clear();
+                operand.setType(ExpressionType.FUNCTION);
+                operand.setFunctionCall(getJsonExtractFunction(parts, DataSchema.ColumnDataType.JSON));
+              }
+            }
+          }
+          break;
+        }
+      }
+    }
+  }
+
+  /**
+   *  @return A string array containing all the parts of an identifier. An identifier may have one or more parts that
+   *  are joined together using <DOT>. For example the identifier "testTable.jsonColumn.name.first" consists up of
+   *  "testTable" (name of table), "jsonColumn" (name of column), "name" (json path), and "first" (json path). The last
+   *  two parts when joined together (name.first) represent a JSON path expression.
+   */
+  private static String[] getIdentifierParts(Identifier identifier) {
+    return identifier.getName().split("\\.");
```

Comment:
Use `StringUtils.split(identifier.getName(), '.')` to avoid the regex match for better performance

Reply:
That won't work, because the second argument is a regex pattern string and hence it will take '.' to mean any character.

Reply:
Fixed.